File: packages/core/src/routes/integrity.ts
/**
 * Integrity routes — content drift detection.
 */

import { Hono } from 'hono'
import type { Services } from '../services'
import { mapStatusCode } from './server'

export function integrityRoutes(services: Services) {
  const app = new Hono()

  app.get('/integrity/drift', (c) => {
    const result = services.integrity.getDriftedEntities()
    if (!result.ok) return c.json({ error: result.error }, mapStatusCode(result.error.code))
    return c.json({ data: result.data })
  })

  return app
}

File: crates/forge-server/src/routes/integrity.rs
//! Integrity routes — content drift detection.
//!
//! Mirrors `packages/core/src/routes/integrity.ts`.

use axum::extract::State;
use axum::routing::get;
use axum::{Json, Router};

use forge_sdk::services::{DriftedEntity, IntegrityService};

use crate::db::with_conn;
use crate::error::ApiError;
use crate::response::ApiData;
use crate::state::SharedState;

// ── Handlers ────────────────────────────────────────────────────────

async fn get_drifted_entities(
    State(state): State<SharedState>,
) -> Result<Json<ApiData<Vec<DriftedEntity>>>, ApiError> {
    let result = with_conn(&state, move |conn| {
        IntegrityService::get_drifted_entities(conn)
    })
    .await?;
    Ok(Json(ApiData { data: result }))
}

// ── Router ──────────────────────────────────────────────────────────

pub fn router() -> Router<SharedState> {
    Router::new()
        .route("/integrity/drift", get(get_drifted_entities))
}
